### ISSUE TITLE
[BUG] ServiceBus receiver refuses to accept messages with unbounded TTL

### ISSUE THREAD

**Issue by aerisnju** [Azure/azure-sdk-for-java#17444]
**Describe the bug**
Bug found in SDK: `microsoft-azure-servicebus 3.5.0`  (**not** `azure-messaging-servicebus`)

The service bus receiver refuses to accept messages with unbounded TTL, producing `IllegalArgumentException` exceptions.

We can create a queue with unbounded (unlimited, -1) TTL. So a message may not have a positive TTL. When the client receives such a message, it refuses to accepted the message because it expects a positive TTL.

This issue was introduced in #13586. I suggest rollback this change or fix the TTL check when receiving a message.

***Exception or Stack Trace***
```
java.lang.IllegalArgumentException: timeToLive must be positive duration.	
at com.microsoft.azure.servicebus.Message.setTimeToLive(Message.java:181) ~[azure-servicebus-3.5.0.jar:?]	
at com.microsoft.azure.servicebus.MessageConverter.convertAmqpMessageToBrokeredMessage(MessageConverter.java:134) ~[azure-servicebus-3.5.0.jar:?]	
at com.microsoft.azure.servicebus.MessageConverter.convertAmqpMessageToBrokeredMessage(MessageConverter.java:95) ~[azure-servicebus-3.5.0.jar:?]	
at com.microsoft.azure.servicebus.MessageReceiver.lambda$receiveAsync$12(MessageReceiver.java:474) ~[azure-servicebus-3.5.0.jar:?]	
at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642) ~[?:?]	
at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]	
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]	
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]	
at java.lang.Thread.run(Thread.java:834) [?:?]
```

**To Reproduce**
1. Create a queue with unbounded TTL.
2. Write a simple test program using microsoft-azure-servicebus version 3.5.0 (3.4.0 does not have such issue) to receive messages from the queue.
3. Send a message without TTL set
4. The program is unable to receive a single message but producing many `IllegalArgumentException`.

***Code Snippet***
N.A. Just create a `IMessageReceiver` and call its `receive()` or `receiveAsync()`.

**Expected behavior**
Messages are received successfully.

**Screenshots**
N.A.

**Setup (please complete the following information):**
 - OS: Ubuntu 2004 in docker (OS independent)
 - IDE : IDE independent
 - Library: `microsoft-azure-servicebus 3.5.0`

**Additional context**
N.A.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
@yvgopal could you please assist with this service bus v3 issue?

/cc @hemanttanwar

**Comment by yvgopal** [comment]
I will be investigating this.

**Comment by yvgopal** [comment]
@aerisnju How are you creating a queue with negative ttl? what is the SDK you are using to create that queue?

**Comment by yvgopal** [comment]
Fix is coming in #17678. will soon release a new version of the SDK.
Silly bug. Any of our tests should have caught it. I will verify why our tests didn't run for that commit which introduced the regression.

**Comment by aerisnju** [comment]
@yvgopal The TTL is `TimeSpan.MaxValue` in C#. For servicebus standard and premium layer this is the default value.
It cannot be set directly in Azure portal but can be set by management client.

**Comment by yvgopal** [comment]
Just released version 3.5.1 with the fix. It should be available for download in the next few hours.
